docs/rules/strict.md

```diff
@@ -139,6 +139,22 @@ function foo() {
 }());
 ```
 
+```js
+/*eslint strict: ["error", "function"]*/
+/*eslint-env es6*/
+
+// Illegal "use strict" directive in function with non-simple parameter list.
+// This is a syntax error since ES2016.
+function foo(a = 1) {
+    "use strict";
+}
+
+// We cannot write "use strict" directive in this function.
+// So we have to wrap this function with a function with "use strict" directive.
```

Comment:
Anyone not reading this example will be aware of this fringe case of function-scoped `'use strict'` when the function has non-simple args. Would it be worth making a mention of this "gotcha" earlier in the document (maybe near the CommonJS note in the opening section) and then linking to the examples section that illustrates the problem?